modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -157,13 +159,90 @@ export default class GeoJsonLayer extends CompositeLayer {
     const LineStringsLayer = this.getSubLayerClass('line-strings', PathLayer);
     const PointsLayer = this.getSubLayerClass('points', ScatterplotLayer);
 
+    const layerProps = {
+      points: {data: undefined, _dataDiff: undefined},
+      lines: {data: undefined, _dataDiff: undefined},
+      polygons: {data: undefined, _dataDiff: undefined},
+      polygonsOutline: {data: undefined, _dataDiff: undefined}
+    };
+
+    const isBinary = this._isBinary();
+
+    if (isBinary) {
+      const {points, lines, polygons} = this.props.data;
+
+      layerProps.points.data = {
+        length: points.positions.value.length / points.positions.size,
+        attributes: {
+          getPosition: points.positions,
+          getProperties: points.properties,
+          getNumericProps: points.numericProps,
```

Comment:
These are not known attributes in the sub-layer. They should be taken out of `attributes` to avoid confusion:

```js
      data: {
        length: points.positions.value.length / points.positions.size,
        attributes: {
          getPosition: points.positions,
        },
        properties: points.properties,
        numericProps: points.numericProps,
        globalFeatureIds: points.globalFeatureIds
      }
```